appinventor/appengine/src/com/google/appinventor/client/editor/simple/components/MockDataFile.java

```diff
@@ -0,0 +1,189 @@
+package com.google.appinventor.client.editor.simple.components;
+
+import com.google.appinventor.client.ErrorReporter;
+import com.google.appinventor.client.Ode;
+import com.google.appinventor.client.editor.simple.SimpleEditor;
+import com.google.gwt.user.client.rpc.AsyncCallback;
+import com.google.gwt.user.client.ui.Image;
+
+import java.util.*;
+
+import static com.google.appinventor.client.Ode.MESSAGES;
+
+public class MockDataFile extends MockNonVisibleComponent {
+  public static final String TYPE = "DataFile";
+
+  private static final String PROPERTY_NAME_SOURCE_FILE = "SourceFile";
+
+  private String sourceFile;
+
+  private List<String> columnNames; // First row of the Data File's contents
+  private List<List<String>> columns; // Parsed columns of the Data File
+  private Set<DataFileChangeListener> dataFileChangeListeners;
+
+  /**
+   * Creates a new instance of a DataFile component whose icon is
+   * loaded dynamically (not part of the icon image bundle)
+   */
+  public MockDataFile(SimpleEditor editor, String type, Image iconImage) {
+    super(editor, type, iconImage);
```

Comment:
TODO: Change MockDataFile icon?

Reply:
@ewpatton I've introduced a new icon for the Data File to prevent ambiguities between the DataFile and the File. Let me know if it's stylistically in tact with the other icons!